scim-server/scim-server-common/src/main/java/org/apache/directory/scim/server/rest/WebApplicationExceptionMapper.java

```diff
@@ -0,0 +1,47 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+
+* http://www.apache.org/licenses/LICENSE-2.0
+
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.directory.scim.server.rest;
+
+import org.apache.directory.scim.spec.protocol.Constants;
+import org.apache.directory.scim.spec.protocol.data.ErrorResponse;
+
+import javax.enterprise.inject.Alternative;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+@Produces(Constants.SCIM_CONTENT_TYPE)
+@Alternative
+public class WebApplicationExceptionMapper implements ExceptionMapper<WebApplicationException> {
```

Comment:
The ExceptionMapper from `commons-jaxrs` is fairly limited in scope (only a handful of supported exception types) and creates a non-scim message body.

The problem is that it is loaded by default by the container.  Marking this class as an `@Alternative` fixes this issue and works with Apache CXF, but order is not guaranteed in other containers.

My suggestion would be remove the conflicting `@Provider` annotation on the impl in `commons-jaxrs` as it forces the use of this component downstream.

Any other ideas on this one?

Reply:
This is a more portable option:

```java
@Provider
@Specializes
@Produces(Constants.SCIM_CONTENT_TYPE)
public class WebApplicationExceptionMapper extends edu.psu.swe.commons.jaxrs.exceptions.mappers.WebApplicationExceptionMapper implements ExceptionMapper<WebApplicationException> {
```

This allows overriding the base class and should be container agnostic!
I'll push this into the PR in the near future

Reply:
Moved this part out to #17